config/jobs/kubernetes/kops/kops-periodics-misc2.yaml

```diff
@@ -300,7 +300,7 @@ periodics:
           -v 2 \
           --up --down \
           --cloud-provider=aws \
-          --create-args="--image='099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20211021' --channel=alpha --networking=kubenet --container-runtime=containerd --api-loadbalancer-type=public --override=cluster.spec.serviceAccountIssuerDiscovery.discoveryStore=s3://k8s-kops-prow/e2e-dc69f71486-5831d.test-cncf-aws.k8s.io/discovery --override=cluster.spec.serviceAccountIssuerDiscovery.enableAWSOIDCProvider=true --override=cluster.spec.iam.useServiceAccountExternalPermissions=true" \
+          --create-args="--image='099720109477/ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-20211021' --channel=alpha --networking=kubenet --container-runtime=containerd --api-loadbalancer-type=public --override=cluster.spec.serviceAccountIssuerDiscovery.discoveryStore=s3://k8s-kops-prow/e2e-kops-grid-scenario-service-account-iam/discovery --override=cluster.spec.serviceAccountIssuerDiscovery.enableAWSOIDCProvider=true --override=cluster.spec.iam.useServiceAccountExternalPermissions=true" \
```

Comment:
Is this change to move the discovery objects outside of the configBase intentional? This means `kops delete cluster` will no longer delete them which may lead to strange behavior if a prow job run inadvertently uses a previous run's objects.

Reply:
I'm trying to put them somewhere that is both specific to the job (so concurrent jobs don't step on each other) and calculatable by that function in build_jobs.py.

Normally the discovery objects would not be inside of configBase since they have significantly different access control. They normally shouldn't even be in the same bucket as the configBase.

I don't think there's much practical risk of a job using a previous run's discovery objects, but perhaps we should make sure delete cluster deletes the discovery objects.